cloudinit/net/dhcp.py

```diff
@@ -791,8 +791,9 @@ def parse_dhcpcd_lease(lease_dump: str, interface: str) -> Dict:
         try:
             lease = dict(
                 [
-                    a.split("=")
+                    a.split("=", maxsplit=1)
                     for a in lease_dump.strip().replace("'", "").split("\n")
+                    if "=" in a
                 ]
             )
         except ValueError as error:
```

Comment:
Can we also check that we've actually parsed some non-empty valid lease config after this try/except block?
```
    if not lease:
            msg = "No valid DHCP lease configuration found in dhcpcd lease"
            LOG.error(msg)
            raise NoDHCPLeaseError(msg)
```

If we proceed beyond this with no valid data, we'll only be injecting a worthless `interface: ens5` in our dhcp lease, which does not provide cloud-init with actionable config anyway.

Reply:
Can we also add the `lease_dump` to the ValueError `LOG.error` message? The reason being that there is not enough information provided in the error level message beyond the actually dict comprehension traceback to tell us what content we failed to parse.  Even though the source dump_lease content is logged as a `debug` level before, some log aggregators (and cloud-init status --format=json) will ignore debug level messages forcing a triager to jump into the full cloud-init log to better understand the dhcp lease content or absence of content which was unable to be parsed.